### ISSUE TITLE
MacOS CI builds fail in time tests

### ISSUE THREAD

**Issue by alt-graph** [gul-cpp/gul14#13]
Apparently GitHub's CI runners are slower that what we saw before, so many of our time-related tests fail:
```
Log of Meson test suite run on 2022-08-19T10:15:25.885403

Inherited environment: GITHUB_JOB=build-and-run-tests-posix GITHUB_EVENT_PATH=/Users/runner/work/_temp/_github_workflow/event.json GITHUB_TRIGGERING_ACTOR=alt-graph RUNNER_OS=macOS XCODE_12_DEVELOPER_DIR=/Applications/Xcode_12.5.1.app/Contents/Developer GITHUB_BASE_REF='' NVM_CD_FLAGS='' ANDROID_HOME=/Users/runner/Library/Android/sdk SHELL=/bin/bash CHROMEWEBDRIVER=/usr/local/Caskroom/chromedriver/104.0.5112.79 GOROOT_1_17_X64=/Users/runner/hostedtoolcache/go/1.17.13/x64 GITHUB_REF_NAME=ci/add_github_ci PIPX_BIN_DIR=/usr/local/opt/pipx_bin GITHUB_REPOSITORY_OWNER=gul-cpp TMPDIR=/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/ RUNNER_ARCH=X64 GITHUB_RUN_ATTEMPT=1 GITHUB_RUN_NUMBER=27 GITHUB_ACTIONS=true ANDROID_SDK_ROOT=/Users/runner/Library/Android/sdk RUNNER_WORKSPACE=/Users/runner/work/gul14 GITHUB_REF_PROTECTED=false RCT_NO_LAUNCH_PACKAGER=1 JAVA_HOME_8_X64=/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.345-1/x64/Contents/Home/ RUNNER_PERFLOG=/usr/local/opt/runner/perflog NUNIT_BASE_PATH=/Library/Developer/nunit GITHUB_WORKFLOW='CI for GUL14' GITHUB_REF=refs/heads/ci/add_github_ci LC_ALL=en_US.UTF-8 NUNIT3_PATH=/Library/Developer/nunit/3.6.0 JAVA_HOME_11_X64=/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/11.0.16-8/x64/Contents/Home/ RUNNER_TOOL_CACHE=/Users/runner/hostedtoolcache GITHUB_ACTION_REPOSITORY='' NVM_DIR=/Users/runner/.nvm USER=runner GITHUB_REF_TYPE=branch GITHUB_API_URL=https://api.github.com GITHUB_EVENT_NAME=push GITHUB_SHA=f10aaf1f06f7dc08a89df47145fdbafef3cbb4e9 RUNNER_TEMP=/Users/runner/work/_temp RUNNER_NAME='Hosted Agent' ANDROID_NDK_ROOT=/Users/runner/Library/Android/sdk/ndk/25.0.8775105 ANDROID_NDK_LATEST_HOME=/Users/runner/Library/Android/sdk/ndk/25.0.8775105 ImageVersion=20220815.1 SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.P5BBfDhRCK/Listeners GITHUB_SERVER_URL=https://github.com __CF_USER_TEXT_ENCODING=0x1F5:0:0 HOMEBREW_NO_AUTO_UPDATE=1 GITHUB_HEAD_REF='' AGENT_TOOLSDIRECTORY=/Users/runner/hostedtoolcache GITHUB_GRAPHQL_URL=https://api.github.com/graphql JAVA_HOME_17_X64=/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.4-8/x64/Contents/Home/ GOROOT_1_19_X64=/Users/runner/hostedtoolcache/go/1.19.0/x64 PATH=/usr/local/lib/ruby/gems/2.7.0/bin:/usr/local/opt/ruby@2.7/bin:/usr/local/opt/pipx_bin:/Users/runner/.cargo/bin:/usr/local/opt/curl/bin:/usr/local/bin:/usr/local/sbin:/Users/runner/bin:/Users/runner/.yarn/bin:/Users/runner/Library/Android/sdk/tools:/Users/runner/Library/Android/sdk/platform-tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/usr/bin:/bin:/usr/sbin:/sbin:/Users/runner/.dotnet/tools:/Users/runner/.ghcup/bin:/Users/runner/hostedtoolcache/stack/2.7.5/x64 _=./meson.py GITHUB_RETENTION_DAYS=90 PERFLOG_LOCATION_SETTING=RUNNER_PERFLOG CONDA=/usr/local/miniconda VM_ASSETS=/usr/local/opt/runner/scripts PWD=/Users/runner/work/gul14/gul14 DOTNET_ROOT=/Users/runner/.dotnet EDGEWEBDRIVER=/usr/local/share/edge_driver JAVA_HOME=/Users/runner/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.345-1/x64/Contents/Home/ VCPKG_INSTALLATION_ROOT=/usr/local/share/vcpkg LANG=en_US.UTF-8 XCODE_13_DEVELOPER_DIR=/Applications/Xcode_13.2.1.app/Contents/Developer ImageOS=macos11 XPC_FLAGS=0x0 PIPX_HOME=/usr/local/opt/pipx GITHUB_ACTOR=alt-graph GECKOWEBDRIVER=/usr/local/opt/geckodriver/bin XPC_SERVICE_NAME=0 HOME=/Users/runner SHLVL=2 RUNNER_TRACKING_ID=github_6de8b030-fc1c-48cb-a5b3-500884998b75 GITHUB_WORKSPACE=/Users/runner/work/gul14/gul14 GITHUB_ACTION_REF='' CI=true GRAALVM_11_ROOT=/Library/Java/JavaVirtualMachines/graalvm-ce-java11-22.2.0/Contents/Home/bin GITHUB_RUN_ID=2888690823 LOGNAME=runner GITHUB_ENV=/Users/runner/work/_temp/_runner_file_commands/set_env_a210191e-dd58-4132-b4c3-6881ef53585f LC_CTYPE=en_US.UTF-8 HOMEBREW_CLEANUP_PERIODIC_FULL_DAYS=3650 HOMEBREW_CASK_OPTS=--no-quarantine POWERSHELL_DISTRIBUTION_CHANNEL=GitHub-Actions-macos11 GOROOT_1_18_X64=/Users/runner/hostedtoolcache/go/1.18.5/x64 ANDROID_NDK_HOME=/Users/runner/Library/Android/sdk/ndk/25.0.8775105 ANDROID_NDK=/Users/runner/Library/Android/sdk/ndk/25.0.8775105 BOOTSTRAP_HASKELL_NONINTERACTIVE=1 GITHUB_STEP_SUMMARY=/Users/runner/work/_temp/_runner_file_commands/step_summary_a210191e-dd58-4132-b4c3-6881ef53585f XCODE_11_DEVELOPER_DIR=/Applications/Xcode_11.7.app/Contents/Developer GITHUB_REPOSITORY=gul-cpp/gul14 GITHUB_PATH=/Users/runner/work/_temp/_runner_file_commands/add_path_a210191e-dd58-4132-b4c3-6881ef53585f STATS_PFS=true GITHUB_ACTION=__run_7 DOTNET_MULTILEVEL_LOOKUP=0 

==================================== 1/1 =====================================
test:         all
start time:   10:15:25
duration:     5.69s
result:       exit status 7
command:      MALLOC_PERTURB_=191 /Users/runner/work/gul14/gul14/build.asan/tests/libgul-test
----------------------------------- stdout -----------------------------------

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libgul-test is a Catch v2.13.7 host application.
Run with -? for options

-------------------------------------------------------------------------------
Scenario: After tic() and sleep(), toc() yields the correct time span
       When: two 50 ms delays are made via sleep()
       Then: toc() measures approximately 50 ms after the first one
-------------------------------------------------------------------------------
../tests/test_time_util.cc:55
...............................................................................

../tests/test_time_util.cc:61: FAILED:
  REQUIRE( toc_s < 0.050 + S_AFTER )
with expansion:
  0.132904987 < 0.0680000011

-------------------------------------------------------------------------------
Scenario: After tic() and sleep(), toc() yields the correct time span
       When: two 50 ms delays are made via sleep()
       Then: toc() measures approximately 100 ms after the second one
-------------------------------------------------------------------------------
../tests/test_time_util.cc:68
...............................................................................

../tests/test_time_util.cc:74: FAILED:
  REQUIRE( toc_s < 0.1 + 2 * S_AFTER )
with expansion:
  0.173287515 < 0.1360000022

-------------------------------------------------------------------------------
Scenario: After tic() and sleep(), toc() yields the correct time span
       When: a 50 ms delay is made via sleep()
       Then: toc() measures approximately 50 ms afterwards
-------------------------------------------------------------------------------
../tests/test_time_util.cc:84
...............................................................................

../tests/test_time_util.cc:90: FAILED:
  REQUIRE( toc_s < 0.05 + S_AFTER )
with expansion:
  0.164803243 < 0.0680000011

-------------------------------------------------------------------------------
Scenario: sleep(..., interrupt) can be interrupted from another thread
       When: interrupting sleep(2s, interrupt) after 15 ms
       Then: the elapsed time is approximately 15 ms
-------------------------------------------------------------------------------
../tests/test_time_util.cc:207
...............................................................................

../tests/test_time_util.cc:213: FAILED:
  REQUIRE( toc_s < 0.015 + S_AFTER )
with expansion:
  0.125236673 < 0.0330000011

-------------------------------------------------------------------------------
Scenario: Trigger::wait*() resumes if another thread calls trigger()
      Given: a trigger and a helper thread that sets it to true after 30 ms
       When: the main thread calls wait_for(1s)
       Then: it resumes after approximately 50 ms
-------------------------------------------------------------------------------
../tests/test_Trigger.cc:89
...............................................................................

../tests/test_Trigger.cc:92: FAILED:
  REQUIRE( time_ms <= 50 + MS_AFTER )
with expansion:
  148 <= 68

-------------------------------------------------------------------------------
Scenario: Trigger::wait*() resumes if another thread calls trigger()
      Given: a trigger and a helper thread that sets it to true after 30 ms
       When: the main thread calls wait_until(now + 1s)
       Then: it resumes after approximately 50 ms
-------------------------------------------------------------------------------
../tests/test_Trigger.cc:101
...............................................................................

../tests/test_Trigger.cc:104: FAILED:
  REQUIRE( time_ms <= 50 + MS_AFTER )
with expansion:
  119 <= 68

-------------------------------------------------------------------------------
Scenario: Trigger::wait_for() and wait_until() wait the requested amount of
          time
       When: calling wait_for(50ms)
       Then: execution resumes after approximately 50 ms
-------------------------------------------------------------------------------
../tests/test_Trigger.cc:121
...............................................................................

../tests/test_Trigger.cc:124: FAILED:
  REQUIRE( time_ms <= 50 + MS_AFTER )
with expansion:
  167 <= 68

===============================================================================
test cases:   333 |   329 passed | 4 failed
assertions: 25398 | 25391 passed | 7 failed

==============================================================================
```

**Comment by alt-graph** [comment]
Ouch... The last PR solved the problems in test_time_util.cc, but I forgot about test_Trigger.cc:
```
-------------------------------------------------------------------------------
Scenario: Trigger::wait_for() and wait_until() wait the requested amount of
          time
       When: calling wait_until(now + 50ms)
       Then: execution resumes after approximately 50 ms
-------------------------------------------------------------------------------
../tests/test_Trigger.cc:133
...............................................................................

../tests/test_Trigger.cc:136: FAILED:
  REQUIRE( time_ms <= 50 + MS_AFTER )
with expansion:
  74 <= 68

===============================================================================
test cases:   337 |   336 passed | 1 failed
assertions: 25493 | 25492 passed | 1 failed
==============================================================================
```
I am going to prepare another PR.

**Comment by alt-graph** [comment]
The issue with `Trigger` is addressed in #15.

**Comment by alt-graph** [comment]
Resolved!
